test/ucp/storetest/shared.go

```diff
@@ -397,87 +425,65 @@ func RunTest(t *testing.T, client store.StorageClient, clear func(t *testing.T))
 		require.NoError(t, err)
 
 		t.Run("query_resources_at_resource_group_scope", func(t *testing.T) {
-			objs, err := client.Query(ctx, store.Query{RootScope: ResourceGroup1Scope})
+			objs, err := client.Query(ctx, store.Query{RootScope: ResourceGroup1Scope, ResourceType: NestedResourceType1})
 			require.NoError(t, err)
 			expected := []store.Object{
-				obj1,
 				nested1,
+				nested2,
+				nested3,
+				nested4,
 			}
 			CompareObjectLists(t, expected, objs.Items)
 		})
 
 		t.Run("query_resources_at_resource_group_scope_with_field_filter", func(t *testing.T) {
-			filters := []store.QueryFilter{{Field: "value", Value: "1"}}
-			objs, err := client.Query(ctx, store.Query{RootScope: ResourceGroup1Scope, Filters: filters})
-			require.NoError(t, err)
-			expected := []store.Object{
-				obj1,
-			}
-			CompareObjectLists(t, expected, objs.Items)
-		})
-
-		t.Run("query_resources_at_resource_group_scope_with_prefix", func(t *testing.T) {
```

Comment:
You'll notice a lot of tests were removed - about 40% of the test cases for Query. That's a **good thing**. We removed the need for a lot of these.